src/gtest/test_history.cpp

```diff
@@ -169,3 +169,37 @@ TEST(History, EpochBoundaries) {
     EXPECT_EQ(view.GetHistoryLength(2), 0);
 
 }
+
+TEST(History, GarbageMemoryHash) {
+    const auto consensusBranchId = NetworkUpgradeInfo[Consensus::UPGRADE_HEARTWOOD].nBranchId;
+
+    FakeCoinsViewDB fakeDB;
+    CCoinsViewCache view(&fakeDB);
+
+    // Hash two history nodes
+    HistoryNode node0 = getLeafN(1);
+    HistoryNode node1 = getLeafN(2);
+
+    view.PushHistoryNode(consensusBranchId, node0);
+    view.PushHistoryNode(consensusBranchId, node1);
+
+    uint256 historyRoot = view.GetHistoryRoot(consensusBranchId);
+
+    // Change garbage memory and re-hash nodes
+    FakeCoinsViewDB fakeDBGarbage;
+    CCoinsViewCache viewGarbage(&fakeDBGarbage);
```

Comment:
How do we know that this has different garbage (or rather, that it would have had different garbage before we changed it to clear that memory)?